src/main/java/uk/gov/pay/api/utils/CustomDateDeserializer.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.pay.api.utils;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class CustomDateDeserializer extends StdDeserializer<ZonedDateTime> {
+
+    private static SimpleDateFormat formatter
+            = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
```

Comment:
Yikes! `SimpleDateFormatter` isn’t thread-safe!

(I know because of, erm, previous experience with this. Yes, in a production system. And since you ask, yes, it did corrupt data stored client-side on users’ machines.)

I think we have employees younger than `SimpleDateFormat`. Use something from the `java.time` package, which has thread-safe formatters.

Again, I’m surprised we don’t have something like this already.

Reply:
Removed.